src/NuGet.Core/NuGet.PackageManagement/Resolution/ResolverGather.cs

```diff
@@ -533,13 +535,37 @@ private void QueueWork(IReadOnlyList<SourceResource> sources, string packageId,
 
         private void QueueWork(IReadOnlyList<SourceResource> sources, PackageIdentity package, bool ignoreExceptions, bool isInstalledPackage)
         {
+            IReadOnlyList<string> configuredPackageSources = null;
+
+            if (_isPackageNamespaceEnabled)
```

Comment:
You can add tests for this in https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.PackageManagement.Test/ResolverGatherTests.cs as well.

Reply:
Added, please check.